lib/index.js

```diff
@@ -94,6 +94,11 @@ Server.prototype.checkRequest = function(req, fn) {
 Server.prototype.serveClient = function(v){
   if (!arguments.length) return this._serveClient;
   this._serveClient = v;
+
+  if(v && !clientSource) {
+      clientSource = read(require.resolve('socket.io-client/socket.io.js'), 'utf-8');
```

Comment:
I apologise for not explaining my idea.

I didn't mean to just move those lines to the the  [Server's .serve() method](https://github.com/codigi/socket.io/blob/a9c9b3dcc8afe890b406880be7f65906e4b5853a/lib/index.js#L291-L294).

What I implying to do something similar to [pillarjs/send](https://github.com/pillarjs/send) (the **streaming** static file server that express's [serve-static](https://github.com/expressjs/serve-static)) is built on.

An example for this would look like:
```js
// In the top of index.js
var fs = require('fs');
var clientPath;

Server.prototype.serve = function(req, res){
  var etag = req.headers['if-none-match'];
  if (etag) {
     // Use cached version.
  }

  res.setHeader('Content-Type', 'application/javascript; charset=utf-8');
  res.setHeader('ETag', clientVersion);
  res.writeHead(200);

  clientPath = clientPath || require.resolve('socket.io-client/socket.io.js');
  var readStream = fs.createReadStream(clientPath);
  readStream.pipe(res);
};
```

Serving static files this way has multiple benefits:

1. The file doesn't unnecessary consume memory if it is not requested.

2. Instead of buffering the entire file into memory, it is **asynchronously** streamed in small chunks (usually 64kb) to the response.

3. The file's chunks can be garbage collected once they were served (which can save memory if it is not accessed frequently).

4. We avoid unnecessary converting the `UTF-8` binary into a string and vise-versa when the client source is flushed to the response.